recipes/libcurl/all/test_package/CMakeLists.txt

```diff
@@ -5,3 +5,7 @@ find_package(CURL REQUIRED)
 
 add_executable(${PROJECT_NAME} test_package.c)
 target_link_libraries(${PROJECT_NAME} CURL::libcurl)
+if(APPLE)
+  find_library(CORE_FOUNDATION CoreFoundation)
+  target_link_libraries(${PROJECT_NAME} ${CORE_FOUNDATION})
```

Comment:
This should be in the recipe There's a cpp_info frameworks thingy

https://github.com/conan-io/conan-center-index/blob/266ec62926ef947a73a35f52eada7a48ca70f378/recipes/libgettext/all/conanfile.py#L160